Simplified/Reader2/NYPLReaderBookmarksBusinessLogic.swift

```diff
@@ -143,15 +149,28 @@ class NYPLReaderBookmarksBusinessLogic: NSObject, NYPLReadiumViewSyncManagerDele
     bookmarks.append(bookmark)
     bookmarks.sort { $0.progressWithinBook < $1.progressWithinBook }
 
-    /*
-     // TODO: SIMPLY-2804 sync with server
-     // this is what happens in R1
-     [self.delegate updateCurrentBookmark:bookmark];
-     [self.syncManager addBookmark:bookmark withCFI:bookmark.location forBook:book.identifier];
-     */
+    addBookmark(bookmark)
 
     return bookmark
   }
+    
+  private func addBookmark(_ bookmark: NYPLReadiumBookmark) {
+    guard
+      let currentAccount = currentLibraryAccountProvider.currentAccount,
+      let accountDetails = currentAccount.details,
+      accountDetails.syncPermissionGranted else {
+        self.bookRegistryProvider.add(bookmark, forIdentifier: book.identifier)
+        return
+    }
+    
+    NYPLAnnotations.postBookmark(forBook: book.identifier,
+                                 toURL: nil,
+                                 bookmark: bookmark) { (serverAnnotationID) in
+      Log.debug(#function, serverAnnotationID != nil ? "Bookmark upload succeed" : "Bookmark failed to upload")
```

Comment:
in case of failure, should we log an event with NYPLErrorLogger instead ? I don't think `postBookmark(forBook...` logs it. Or we could log it there, since I think you said it does do that for deletions.

Reply:
`postBookmark(forBook...` does log the event with `Log.error()` like the deletions. The actual network request happens in `postAnnotation(forBook...` and that's where we have access to the network error. Maybe another ticket/PR to update the whole `NYPLAnnotations` class to log with NYPLErrorLogger when possible?

Reply:
sounds good, make sure to file the ticket. Keep in mind that logging with `Log.error` alone does pretty much nothing -- in the sense that that log will be buried in the next event that it is reported.